src/test/java/com/crud/tasks/trello/client/TrelloClientTest.java

```diff
@@ -0,0 +1,86 @@
+package com.crud.tasks.trello.client;
+
+import com.crud.tasks.domain.CreatedTrelloCard;
+import com.crud.tasks.domain.TrelloBoardDto;
+import com.crud.tasks.domain.TrelloCardDto;
+import com.crud.tasks.trello.config.TrelloConfig;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TrelloClientTest {
+
+    @InjectMocks
+    private TrelloClient trelloClient;
+
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Mock
+    private TrelloConfig trelloConfig;
+
+    @Before
+    public void init() {
+        when(trelloConfig.getTrelloApiEndpoint()).thenReturn("http://test.com");
+        when(trelloConfig.getTrelloAppKey()).thenReturn("test");
+        when(trelloConfig.getTrelloToken()).thenReturn("test");
+    }
+
+    @Test
+    public void shouldFetchTrelloBoards() throws URISyntaxException {
+        // Given
+        TrelloBoardDto[] trelloBoards = new TrelloBoardDto[1];
+        trelloBoards[0] = new TrelloBoardDto("test_id", "test_board", new ArrayList<>());
+
+        URI uri = new URI("http://test.com/members/pb_kodilla/boards?key=test&token=test&fields=name,id&lists=all");
+
+        when(restTemplate.getForObject(uri, TrelloBoardDto[].class)).thenReturn(trelloBoards);
+
+        // When
+        List<TrelloBoardDto> fetchedTrelloBoards = trelloClient.getTrelloBoards();
+
+        // Then
+        assertEquals(1, fetchedTrelloBoards.size());
+        assertEquals("test_id", fetchedTrelloBoards.get(0).getId());
+        assertEquals("test_board", fetchedTrelloBoards.get(0).getName());
+        assertEquals(new ArrayList<>(), fetchedTrelloBoards.get(0).getLists());
+    }
```

Comment:
ten test sie wyklada, poniewaz `when(restTemplate.getForObject(uri, TrelloBoardDto[].class)).thenReturn(trelloBoards);` nie dziala

trelloBoard zamockowany bedzie zwrocony jesli uri bedzie rowne
`http://test.com/members/pb_kodilla/boards?key=test&token=test&fields=name,id&lists=all`

u ciebie uri jest nastepujace
`http://test.com/members/null/boards?filter=all&fields=id,name&lists=all&key=test&token=test`

Zeby to zadzialalo musisz zrobic mocka trello config, tak aby zwracal odmowienia wartosc `trelloConfig.getTrelloUsername()`

Reply:
dodatkowo kolejnosc argumentow jest tutaj tez wazna. Aby to zadzialalo, string musi byc dokladnie taki sam